### PR TITLE
GSoC'21 Mid-Term Progress: Aitik Gupta

### PR DESCRIPTION
**Pull request by aitikgupta**
We're almost halfway through!

image

### CODE REVIEW

**Comment by davidevaleriani** [comment]
@story645 do you want to check this before me merging?

File: content/posts/GSoC_2021_MidTerm/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Something which one of my mentors said which _really_ stuck with me:
+> Matplotlib isn't a font library, and shouldn't try to be one.
+
+It's really easy to fall into the trap of trying to do _everything_ within your own project, which ends up rather _hurting_ itself.
+<hr>
+
+To give an analogy, <ins>Google Docs</ins> try to do _a lot_ of things with its platform, from basic text editing to full-fledged project management.
+
+Compared with <ins>Notion</ins>, whose whole _niche_ is to provide a platform for project management (among other things), **some people tend to choose the latter for their specific needs.**
```
Comment (story645) [inline content/posts/GSoC_2021_MidTerm/index.md]: Not sure this metaphor is helping explain your point 'cause you don't go deep enough to explain why gdocs being all things is bad (and not sure you should cause it'll distract)

File: content/posts/GSoC_2021_MidTerm/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+PS: If that font stuff didn't make sense, I would recommend going through a friendly tutorial I wrote, which is all about [Matplotlib and Fonts](https://matplotlib.org/stable/users/fonts.html)!
+## Unexpected Complications
+Matplotlib uses an external dependency `ttconv` which was initially forked into Matplotlib's repository **in 2003**!
+>  ttconv was a standalone commandline utility for converting TrueType fonts to subsetted Type 3 fonts (among other features) written in 1995, which Matplotlib forked in order to make it work as a library.
+
+Over the time, there were a lot of issues with it which were either hard to fix, or didn't attract a lot of attention. (See the above paragraph for a valid reason)
+
+One major utility which is still used is `convert_ttf_to_ps`, which takes a _font path_ as input and converts it into a Type 3 or Type 42 PostScript font, which can be embedded within PS/EPS output documents. The guide I wrote (mentioned just above the recent heading) contains decent descriptions, the differences between these type of fonts, etc.
```
Comment (story645) [inline content/posts/GSoC_2021_MidTerm/index.md]: Maybe just relink the guide?

**Review by story645 — COMMENTED** [review]
Great post, like the balance of detail and whole picture. Couple of nits.

File: content/posts/GSoC_2021_MidTerm/index.md (hunk 1 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_MidTerm/index.md]: Ah, okay. I can remove this whole analogy itself 👍🏼

File: content/posts/GSoC_2021_MidTerm/index.md (hunk 2 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_MidTerm/index.md]: 👍🏼

**Review by story645 — APPROVED** [review]

**Comment by aitikgupta** [comment]
Mind if I bump this up?
I need to add the deployed link to the spreadsheet 😄

**Comment by davidevaleriani** [comment]
Sure, let me know when ready to publish

**Comment by aitikgupta** [comment]
I think it's ready to publish, since @story645 has approved after I worked on her review.

**Comment by story645** [comment]
Yeah sorry only reason I didn't merge is cause every time I try, something usually goes wrong

**Pull request merged**